*: sha256 = sha256-.. => hash = sha256-..

mkg20001 opened the pull request:
This commit replaces all hashes using the new format
and the old variable name (sha256 = "sha256-..") to
the new format (hash = "sha256-..")

Command used for changes

find -iname "*.nix" -type f -exec sed 's|sha256 = "sha256|hash = "sha256|g' -i {} +

//cc @worldofpeace

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented:
This breaks easy updates for people not using sri hashes.

stigtsp commented on pkgs/top-level/perl-packages.nix:
```diff
@@ -10184,7 +10184,7 @@ let
     src = fetchurl {
       url = "mirror://cpan/authors/id/P/PE/PEVANS/IO-Async-0.78.tar.gz";
-      sha256 = "sha256-P7UYhZd7hiGKiepC84yvvOWCO/SPqqLRhaGGwqNYNmw=";
+      hash = "sha256-P7UYhZd7hiGKiepC84yvvOWCO/SPqqLRhaGGwqNYNmw=";
```
```suggestion
      sha256 = "sha256-P7UYhZd7hiGKiepC84yvvOWCO/SPqqLRhaGGwqNYNmw=";
```

stigtsp commented on pkgs/top-level/perl-packages.nix:
```diff
@@ -23090,7 +23090,7 @@ let
     src = fetchurl {
       url = "mirror://cpan/authors/id/G/GR/GRANTM/XML-Filter-Sort-1.01.tar.gz";
-      sha256 = "sha256-UQWF85pJFszV+o1UXpYXnJHq9vx8l6QBp1aOhBFi+l8=";
+      hash = "sha256-UQWF85pJFszV+o1UXpYXnJHq9vx8l6QBp1aOhBFi+l8=";
```
```suggestion
      sha256 = "sha256-UQWF85pJFszV+o1UXpYXnJHq9vx8l6QBp1aOhBFi+l8=";
```

stigtsp reviewed (changes requested):
Hi! The [update script](https://github.com/stigtsp/nix-update-cpan) expects to find and update `sha256`. I suggest keeping `sha256` for the two derivations in `pkgs/top-level/perl-packages.nix` if that's ok?

mkg20001 commented:
@SuperSandro2000 I suppose this PR might possibly be the right time to resolve those issues beforehand and then merge it. I'd say SRI is the future, so we shouldn't let that hold things back.

SuperSandro2000 commented:
This should be done in badges or we can't review it and get frequent merge conflicts. Also right now it does not even eval and what do we do with all the hashes in the old format? If they are recalculated it definitely needs to be done in batches. Does this cause rebuilds? Is there a consensus on this? What do we do with PRs that introduce the old format?

worldofpeace commented:
Anyone can do `nix to-sri --type sha256  0f7xysk0cl48q7i28m25hasmrp30grgm3kah0s7xmkjgm33887pi` @SuperSandro2000

mkg20001 committed:
*: sha256 = sha256-.. => hash = sha256-..

This commit replaces all hashes using the new format
and the old variable name (sha256 = "sha256-..") to
the new format (hash = "sha256-..")

Command used for changes

find -iname "*.nix" -type f -exec sed 's|sha256 = "sha256|hash = "sha256|g' -i {} +

//cc @worldofpeace

jonringer commented:
> Anyone can do `nix to-sri --type sha256 0f7xysk0cl48q7i28m25hasmrp30grgm3kah0s7xmkjgm33887pi` @SuperSandro2000

Yea, but it's not a good user experience. At the very least I don't think we should be requesting changes to PR because they have a `sha256 = <base32 hash>;`.

Once 2.4 is stable, then the changes will mostly be `sha256` -> `hash`, as the sri hash will be default.

jtojnar commented:
Also `fetchpatch` and `fetchgit` (therefore also `fetchFromGitHub` with submodules) do not support hash yet: https://github.com/NixOS/nixpkgs/pull/79987

vcunat commented:
Honestly I fail to see motivation why replace all the hashes.

jtojnar commented:
I think standardization on a single format is a good idea to reduce the confusion for new users but it is simply too soon.

We need to decide if the attribute name should be `narHash`, `hash`, or `integrity` first, and then add support for it  to [`fetchpatch` and `fetchgit`](https://github.com/NixOS/nixpkgs/pull/79987). Only then we will be able to gradually migrate to the new attribute.
